### PR TITLE
Use CodingKey Fix #27

### LINKED ISSUE #27: Report `CodingKey` stuff from IBLinter
To use IBDecodable in IBLinter https://github.com/IBDecodable/IBLinter/issues/38 I remove a lot of code that has been evolved since the creation of IBDecodable 
https://github.com/IBDecodable/IBLinter/commit/fa74537620b0661d3c37ebd1dac96e229b3336e6

I think using enum is a good stuff!
For boolean maybe we need some mapping 'misplaced' -> I want to keen `isMisplaced`

### PR DESCRIPTION
**Pull request by kateinoigakukun**

### CODE REVIEW

**Comment by SD10** [comment]
@kateinoigakukun You need to update `.travis.yml` to use Xcode 9.3 for Swift 4.1

File: Sources/Models/Views/ScrollViews/CollectionView.swift [hunk 1]
```diff
@@ -82,50 +97,74 @@ public struct CollectionViewCell: XMLDecodable, ViewProtocol {
...
+                case .key: return "contentMode"
+                default: return key.stringValue
+                }
+            }()
+            return MappedCodingKey(stringValue: stringValue)
+        }
         return CollectionViewLayout(
-            id:                       try xml.attributeValue(of: "id"),
-            key:                      xml.attributeValue(of: "contentMode"),
-            customClass:              xml.attributeValue(of: "customClass"),
```
Comment (phimage) [inline Sources/Models/Views/ScrollViews/CollectionView.swift]: I think I make an issue here. key is key , no mapping needed
#22

**Comment by phimage** [comment]
I do no read how work exactly `CodingKey` that's why I make the issue
 but for "special" key mapping I am wondering if this code could help to simplify
```swift
enum CodingKeys: String {
    case key
    case misplaced = "isMisplaced"
}

print(CodingKeys.key.rawValue)  "key"
print(CodingKeys.misplaced.rawValue) "isMisplaced"

print(CodingKeys.key) "key"
print(CodingKeys.misplaced) // "misplaced"
```

File: Sources/Models/Views/ScrollViews/CollectionView.swift (hunk 1 above)
Comment (SD10) [reply Sources/Models/Views/ScrollViews/CollectionView.swift]: @phimage So what do you suggest, using a constant like `"collectionViewLayout"`?
Or do you mean the typo for `contentMode -> key`? Cuz it's been fixed

Comment (phimage) [reply Sources/Models/Views/ScrollViews/CollectionView.swift]: If I fix my code
```swift
key:                      xml.attributeValue(of: "key")
```
so `.key`will be used without key mapping

**Comment by kateinoigakukun** [comment]
@phimage Thanks for your suggestion. Your code is more simple than mine but there is an issue. With your idea we have to enumerate **all** properties in `CodingKeys` when more than one mapping is needed. If `CodingKeys` were not generated as private, we can override `stringValue` property like this 🤔.

```swift
extension CollectionViewCell.CodingKeys {
    var stringValue: String {
        switch self {
            case .isMisplaced: return "misplaced"
            default: return String(describe: self)
        }
    }
}
```

But `CodingKeys` generated by swift compiler is enabled to use in private scope, so I use `MappedCodingKey` as stringValue wrapper.

**Comment by phimage** [comment]
Ok there is some limitation with auto generated stuff on enum

**Pull request merged**